generic/piranha-tree-sitter/src/test-resources/java/input/XPFlagCleanerInterfaceMethod.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2022 Uber Technologies, Inc.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.piranha;
+
+import dagger.Provides;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.inject.Inject;
+
+class XPFlagCleanerPositiveCases {
+
```

Comment:
I haven't paid close attention to comparing the input, treatment and control refactorings. Assume, these will be consistent with our existing refactorings. If not, we can iterate subsequently on fixing any open issues.

Reply:
These test cases are exactly like the ones we have for the error prone version.